Docs 7754 dwformats duke (#1319)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-formats-ndjson.adoc

[source,dataweave,linenums]
----
%dw 2.0
var myInput = readUrl('classpath://myInput.ndjson', 'application/x-ndjson, {skipInvalid=true})
output application/json
---
myInput
----

==== Output

The JSON output is an array of the valid objects from the ndjson input.

[source,weave,linenums]
----
[
  {
    "name": "Mariano"
  },
  {
    "name": "Shoki"
  }
]
----

// CONFIG PROPS ///////////////////////////////////////////////////////

[properties]]
== Configuration Properties

DataWeave supports the following configuration properties for the Newline Delimited JSON format (ndjson).

=== Reader Properties

The ndjson reader accepts properties that provide instructions for reading input data.

[cols="1,1,1,3a", options="header"]
|===
| Parameter | Type | Default | Description
| `ignoreEmptyLine` | `Boolean` | `true` | Ignores any empty line. Valid values are `true` or `false`.
| `skipInvalid` | `Boolean` | `false` | Skips data that is not valid ndjson. Valid values are `true` or `false`.
|===

=== Writer Properties

The ndjson writer accepts properties that provide instructions for writing output data.

[cols="1,1,1,3a", options="header"]
|===
| Parameter | Type | Default | Description

> | `bufferSize` | `Number` | `8192` | Size of the buffer writer.

Reviewer: ```suggestion
| `bufferSize` | `Number` | `8192` | Size of the writer buffer.
```

> | `deferred` | `Boolean` | `false` | Property Property for deferred output. Valid values are `true` or `false`.

Reviewer: ```suggestion
| `deferred` | `Boolean` | `false` | Property for deferred output. Valid values are `true` or `false`.
```

> | `encoding` | `String` | None | Encoding for the ndjson reader to use.

Author: ```suggestion
| `encoding` | `String` | `'UTF-8'` | Encoding for the ndjson reader to use.
```

> | `encoding` | `String` | None | Encoding for the ndjson reader to use.

Author: ```suggestion
| `encoding` | `String` | `'UTF-8'` | Encoding for the ndjson reader to use.
```

> | `skipNullOn` | `String` | None | Valid values are `arrays` or `objects`.

Author: ```suggestion
| `skipNullOn` | `String` | `null` | Valid values are `arrays` or `objects`.
```

> | `skipNullOn` | `String` | None | Valid values are `arrays` or `objects`.

Author: ```suggestion
| `skipNullOn` | `String` | `null` | Valid values are `arrays` or `objects`.
```

> | `writeAttributes` | `Boolean` | `false` | Valid values are `true` or `false`.

Author: missing description
